src/bvals/bvals.hpp

```diff
@@ -175,6 +231,7 @@ class BoundaryValues : public BoundaryBase, // public BoundaryPhysics,
   friend class FaceCenteredBoundaryVariable; // needs nface_, nedge_, num_north/south_...
   // TODO(KGF): consider removing these friendship designations:
   friend class CellCenteredBoundaryVariable;
+  friend class BoundarySwarm;
```

Comment:
Ben this isn't your fault, but this many friend classes is symptomatic of a code smell. I think something needs to be redesigned in a separate issue.

Reply:
https://github.com/lanl/parthenon/issues/447

Reply:
I agree with you about the friend classes, I just wanted to go with the existing model for this PR, especially since it's not trivial to get away from this (some useful data about neighbors, for example, is evaluated/stored down in the inheritance/friend chain)